datalake/src/main/java/com/sequenceiq/datalake/controller/sdx/SdxController.java

```diff
@@ -336,7 +338,29 @@ public SdxBackupStatusResponse backupDatalakeStatusByName(@ResourceName String n
     public SdxDatabaseRestoreResponse restoreDatabaseByName(@ResourceName String name, String backupId,
             String restoreId, String backupLocation) {
         SdxCluster sdxCluster = getSdxClusterByName(name);
-        return sdxBackupRestoreService.triggerDatabaseRestore(sdxCluster, backupId, restoreId, backupLocation);
+        try {
+            SdxDatabaseRestoreStatusResponse response = sdxBackupRestoreService.getDatabaseRestoreStatus(sdxCluster, restoreId);
+            SdxDatabaseRestoreResponse sdxDatabaseRestoreResponse = new SdxDatabaseRestoreResponse();
+            sdxDatabaseRestoreResponse.setOperationId(restoreId);
+            return sdxDatabaseRestoreResponse;
+        } catch (NotFoundException notFoundException) {
+            return sdxBackupRestoreService.triggerDatabaseRestore(sdxCluster, backupId, restoreId, backupLocation);
+        }
+    }
+
+    @Override
+    @CheckPermissionByResourceName(action = AuthorizationResourceAction.RESTORE_DATALAKE)
+    public SdxRestoreResponse restoreDatalakeByName(@ResourceName String name, String backupId, String backupLocation) {
+        SdxCluster sdxCluster = getSdxClusterByName(name);
+        return sdxBackupRestoreService.triggerDatalakeRestore(sdxCluster, backupId, backupLocation);
+    }
```

Comment:
API should even accept backup location override. Please change the implementation accordingly.

Reply:
![Screenshot from 2021-06-16 16-21-32](https://user-images.githubusercontent.com/67844271/122296423-ff2ac300-cebf-11eb-9998-407b412c69eb.png)
![Screenshot from 2021-06-16 16-21-17](https://user-images.githubusercontent.com/67844271/122296429-00f48680-cec0-11eb-82fe-5540c8fe76f3.png)
![Screenshot from 2021-06-16 16-28-36](https://user-images.githubusercontent.com/67844271/122296432-02be4a00-cec0-11eb-9a67-608cee59d282.png)

Implemented backuplocationoverride and tested successfully.